packages/twenty-front/src/modules/settings/profile/components/DateTimeSettings.tsx

```diff
@@ -0,0 +1,141 @@
+import styled from '@emotion/styled';
+import { useRecoilState } from 'recoil';
+
+import { currentWorkspaceMemberState } from '@/auth/states/currentWorkspaceMemberState';
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useUpdateOneRecord } from '@/object-record/hooks/useUpdateOneRecord';
+import { DateTimeSettingsDateFormatSelect } from '@/settings/profile/components/DateTimeSettingsDateFormatSelect';
+import { DateTimeSettingsTimeFormatSelect } from '@/settings/profile/components/DateTimeSettingsTimeFormatSelect';
+import { DateTimeSettingsTimeZoneSelect } from '@/settings/profile/components/DateTimeSettingsTimeZoneSelect';
+import { DateFormat } from '@/workspace-member/constants/DateFormat';
+import { TimeFormat } from '@/workspace-member/constants/TimeFormat';
+import { dateTimeFormatState } from '@/workspace-member/states/dateTimeFormatState';
+import { detectDateFormat } from '@/workspace-member/utils/detectDateFormat';
+import { detectTimeFormat } from '@/workspace-member/utils/detectTimeFormat';
+import { detectTimeZone } from '@/workspace-member/utils/detectTimeZone';
+import { getWorkspaceEnumFromDateFormat } from '@/workspace-member/utils/formatDateLabel';
+import { getWorkspaceEnumFromTimeFormat } from '@/workspace-member/utils/formatTimeLabel';
+import {
+  WorkspaceMemberDateFormatEnum,
+  WorkspaceMemberTimeFormatEnum,
+} from '~/generated/graphql';
+import { isDefined } from '~/utils/isDefined';
+import { logError } from '~/utils/logError';
+
+const StyledContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${({ theme }) => theme.spacing(4)};
+`;
+
+export const DateTimeSettings = () => {
+  const [currentWorkspaceMember, setCurrentWorkspaceMember] = useRecoilState(
+    currentWorkspaceMemberState,
+  );
+  const [dateTimeFormat, setDateTimeFormat] =
+    useRecoilState(dateTimeFormatState);
+
+  const { updateOneRecord } = useUpdateOneRecord({
+    objectNameSingular: CoreObjectNameSingular.WorkspaceMember,
+  });
+
+  const updateWorkspaceMember = async (changedFields: any) => {
+    if (!currentWorkspaceMember?.id) {
+      throw new Error('User is not logged in');
+    }
+
+    try {
+      await updateOneRecord({
+        idToUpdate: currentWorkspaceMember.id,
+        updateOneRecordInput: changedFields,
+      });
+    } catch (error) {
+      logError(error);
+    }
+  };
+
+  if (!isDefined(currentWorkspaceMember)) return;
+
+  const handleTimeZoneChange = (timeZone: string) => {
+    const workspaceMember = {
+      ...currentWorkspaceMember,
+      timeZone,
+    };
+    setCurrentWorkspaceMember(workspaceMember);
+
+    setDateTimeFormat({
+      ...dateTimeFormat,
+      timeZone: timeZone === 'system' ? detectTimeZone() : timeZone,
+    });
+
+    updateWorkspaceMember(workspaceMember);
```

Comment:
Getting a 400 error.

<img width="1506" alt="Screenshot 2024-06-25 at 14 20 52" src="https://github.com/twentyhq/twenty/assets/6399865/ba5f6d7d-82b0-4835-90d9-b381cefc6f1e">

Would it be possible to pass just the modified field and not everything everytime? That would solve the above issue and also probably be cleaner